### DISCUSSION TITLE
feature: add section links in pr description

### DISCUSSION THREAD

**Discussion by electriquo** [community/community#127416] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

from [pull request best practices](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/getting-started/best-practices-for-pull-requests#provide-context-and-guidance)
> Write clear titles and descriptions for your pull requests so that reviewers can quickly understand what the pull request does. 

when using headers to format the description, the rendered markdown is missing [section links](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#section-links) in the same as any other rendered markdown file.

it would be awesome to have section links consistent across all rendered markdown and it will allow pull request participate to share a direct link to sections in description.

for example, see this discussion. the section links are missing although there are headers in it :(

**Comment by fmigneault** [comment]
It would be amazing if that was supported natively.

Currently, one has to use some hacks (e.g.: https://gist.github.com/borekb/f83e48479aceaafa43108e021600f7e3) to make a basic Markdown feature work. 

Markdown links work everywhere else such as in rendered `README.md`, so I do not understand why they could not go through the same processing engine when parsing the issue/pull-request descriptions.
